core/src/main/java/hudson/util/RingBufferLogHandler.java

```diff
@@ -86,13 +74,33 @@ public static int getDefaultRingBufferSize() {
     }
 
     @Override
-    public synchronized void publish(LogRecord record) {
-        int len = records.length;
-        records[(start + size) % len] = new LogRecordRef(record);
-        if (size == len) {
-            start = (start + 1) % len;
-        } else {
-            size++;
+    public void publish(LogRecord record) {
+        if (record.getParameters() != null) {
+            try {
+                LogRecord clone = new LogRecord(record.getLevel(), dummyFormatter.formatMessage(record));
```

Comment:
I felt nervous about updating the `LogRecord` in place, though it would probably be thread-safe (since `log` call would consult `Handler`s serially) and would halve the work when combined with `support-core`.